api/hypershift/v1beta1/azure.go

```diff
@@ -422,11 +422,11 @@ type AzurePlatformSpec struct {
 	// +immutable
 	SecurityGroupID string `json:"securityGroupID"`
 
-	// managedIdentities contains the managed identities needed for HCP control plane and data plane components that
-	// authenticate with Azure's API.
+	// azureAuthenticationConfig is the type of Azure authentication configuration to use to authenticate with Azure's
+	// Cloud API.
 	//
 	// +required
-	ManagedIdentities AzureResourceManagedIdentities `json:"managedIdentities"`
+	AzureAuthenticationConfig AzureAuthenticationConfiguration `json:"azureAuthenticationConfig"`
```

Comment:
This should have `omitzero` even if it's required, are you using Go 1.24 yet?

Reply:
Yeah but we can follow up with that I believe - https://github.com/openshift/hypershift/pull/6283#discussion_r2219538345

Reply:
Have you captured following up on this in some sort of story that's been prioritised?

Reply:
>This should have omitzero even if it's required, are you using Go 1.24 yet?
>Yeah but we can follow up with that I believe - #6283 (comment)

I don't  think we can guarantee our consumer clients are on 1.24 hence my suggestion to postpone this.

Reply:
I would expect your Go deps to rely on something `k8s.io` which when updated to 1.33 would require 1.24 as a minimum go version, which will then put the same requirement on your consumers